ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -725,14 +725,33 @@ private predicate singletonMethodOnModule(MethodBase method, string name, Module
     selfInModule(object.(SelfVariableReadAccess).getVariable(), m)
   )
   or
-  flowsToSingletonMethodObject(trackModuleAccess(m), method, name)
+  exists(DataFlow::LocalSourceNode sourceNode |
+    m = resolveConstantReadAccess(sourceNode.asExpr().getExpr()) and
+    flowsToSingletonMethodObject(sourceNode, method, name)
+  )
   or
   exists(Module other |
     extendCallModule(m, other) and
     method = lookupMethod(other, name)
   )
 }
 
+pragma[nomagic]
+private MethodBase lookupSingletonMethod(Module m, string name) {
```

Comment:
Could you add some QLDoc indicating that this takes inheritance into account, and likewise on `singletonMethodOnModule`.

I'd probably have named these predicates in a way that clarifies their relationship, such as:
- `lookupOwnSingletonMethod`
- `lookupSingletonMethod`

Reply:
I simply used the naming after the existing `lookupMethod` predicate for instance methods. I'll add ql doc.